actions.js

```diff
@@ -23,10 +23,10 @@ export function navigate(rawUri, nextFocus = 1, nextContext) {
     if (uri === router.uri) {
       return;
     }
-
     const parsed = await worker.postMessage({
       type: 'parse',
-      uri
+      uri,
+      whitelist: router.routerWhitelist
```

Comment:
we can keep it as whitelist in the state, that sounds alright. But as an incoming parameter I'd keep it named as routerWhitelist (just in case other configurable whitelists might make it in somewhere else)